internal/ingress/annotations/authtls/main.go

```diff
@@ -36,7 +36,7 @@ const (
 
 var (
 	authVerifyClientRegex = regexp.MustCompile(`on|off|optional|optional_no_ca`)
-	commonNameRegex       = regexp.MustCompile(`CN=`)
+	matchCNSanitizer      = regexp.MustCompile(`^[a-zA-Z0-9\.\-_\(\)\|]+$`)
```

Comment:
why underscore? Though underscore shouldn't be part of FQDNs and CNs :D

Can you add an example here on what this regex catches and not? Or maybe in unit tests

Reply:
https://www.digicert.com/kb/ssl-support/underscores-not-allowed-in-fqdns.htm

oh wow, the more you know! Thanks for pointing this out haha I will remove the underscore, honestly I did this a few weeks ago and can't remember why.